client/www/lib/hooks/useEditBlobConstraints.ts

```diff
@@ -0,0 +1,204 @@
+import config from '../config';
+import { jobFetchLoop } from '../indexingJobs';
+
+type JobConstraintTypes = 'require' | 'index' | 'unique' | 'type';
+
+export type PendingJob = {
+  jobType: InstantIndexingJob['job_type'];
+  checkedDataType?: CheckedDataType | null | undefined;
+};
+
+export const useEditBlobConstraints = ({
+  attr,
+  appId,
+  isRequired,
+  isIndexed,
+  isUnique,
+  checkedDataType,
+  token,
+}: {
+  attr: SchemaAttr;
+  appId: string;
+  token: string;
+  isRequired: boolean;
+  isIndexed: boolean;
+  isUnique: boolean;
+  checkedDataType: CheckedDataType | 'any';
+}) => {
+  const [pendingJobs, setPendingJobs] = useState<{
+    [jobType in JobConstraintTypes]?: PendingJob;
+  }>({});
+
+  const [runningjobs, setRunningJobs] = useState<{
+    [jobType in JobConstraintTypes]?: InstantIndexingJob;
+  }>({});
+
+  const fetchLoopsRef = useRef<{ [jobType: string]: { stop: () => void } }>({});
+
+  const [progress, setProgress] = useState<{ [jobType: string]: number }>({});
+
+  // Keep running jobs updated
+  useEffect(() => {
+    const jobTypes = Object.keys(runningjobs) as JobConstraintTypes[];
+
+    for (let i = 0; i < jobTypes.length; i++) {
+      const jobType = jobTypes[i];
+      const job = runningjobs[jobType];
+
+      if (
+        !job ||
+        job.job_status === 'completed' ||
+        job.job_status === 'errored'
+      ) {
+        // Clean up completed/errored jobs
+        if (fetchLoopsRef.current[jobType]) {
+          fetchLoopsRef.current[jobType].stop();
+          delete fetchLoopsRef.current[jobType];
+        }
+        continue;
+      }
+
+      // Skip if already polling this job type
+      if (fetchLoopsRef.current[jobType]) {
+        continue;
+      }
+
+      const fetchLoop = jobFetchLoop(appId, job.id, token);
+      fetchLoopsRef.current[jobType] = fetchLoop;
+
+      fetchLoop.start((updatedJob, error) => {
+        if (error) {
+          return;
+        }
+
+        if (updatedJob) {
+          const workEstimateTotal = updatedJob.work_estimate ?? 50000;
+          const workCompletedTotal = updatedJob.work_completed ?? 0;
+
+          const percent = Math.floor(
+            (workCompletedTotal / workEstimateTotal) * 100,
+          );
+          setProgress((prev) => ({ ...prev, [jobType]: percent }));
+
+          setRunningJobs((prev) => ({
+            ...prev,
+            [jobType]: updatedJob,
+          }));
+        }
+      });
+    }
+  }, [runningjobs, appId, token]);
+
+  useEffect(() => {
+    // If running jobs, don't update any pending
+    const isRunning = Object.values(runningjobs).some(
+      (job) => job.job_status !== 'completed' && job.job_status !== 'errored',
+    );
+    if (isRunning) {
+      return;
+    }
+
+    // Pending requirement job
+    if (isRequired === attr.isRequired) {
+      setPendingJobs((p) => ({ ...p, require: undefined }));
+    } else if (isRequired) {
+      setPendingJobs((p) => ({ ...p, require: { jobType: 'required' } }));
```

Comment:
This made me think: is it possible to get in some kind of race condition where we are running a `required` job, but immediately try to run a `remove-required` job. 

Can something like that happen?